parsec/include/parsec/parsec_config_bottom.h

```diff
@@ -36,6 +36,14 @@
 #define END_C_DECLS            /* empty */
 #endif
 
+/* If we're in C++, redefine restrict to nothing */
+#if defined(c_plusplus) || defined(__cplusplus)
+#  if defined(restrict)
```

Comment:
I suggest to use something along the lines of `parsec_restrict` and not fiddle with reserved keywords.